.pre-commit-config.yaml

```diff
@@ -13,5 +13,7 @@ repos:
         name: gitlint
         entry: gitlint
         stages: [commit-msg]
+      - id: gitlint-ci
+        args: [--commits, "origin/$(git rev-parse --abbrev-ref HEAD)..HEAD"]
```

Comment:
@yftacherzog This makes it check all the commits on the branch.

Reply:
I would expect HEAD points to the last commit on the PR branch and I suspect `origin/$(git rev-parse --abbrev-ref HEAD)` might point to the same point as well. No? (i.e. if the PR branch is foo, then it will run `--commits origin/foo..foo`)

In any case, I don't think we should check all the commits on the branch. Suppose that there was a bad commit message in the past, then it makes us stuck forward. This test should be about what is being introduced on the current PR.

Looking at checks implemented with github workflows in our org, I see them using references that github provides. I think this might be doable here as well: https://github.com/redhat-appstudio/release-service/blob/e2fd1d6ea2ac4e913647dca93b84ba41eacc3f6f/.github/workflows/pr.yml#L123

We have the implementation we did in tekton. It might help as well:
https://github.com/redhat-appstudio/o11y/blob/main/.tekton/pull-request.yaml#L81

Reply:
wasn't aware of this possibility, checking

Reply:
I don't really like maintaining gitlint on pre-commit for local checks and on a separate job for CI, which might create conflicting or missing reports if not handled correctly, but the current solution isn't ideal either. I used your solution for now.

Reply:
I understand. I think the part of being able to tell what's the target branch for the changes in very clear at a PR level, but might not be as clear on a local env. I think this check is ok to run only in CI.

Reply:
Locally there won't be a need to check all commits, as we'll be unable to commit something if it does not pass the gitlint